dashboard/new-dashboard/src/components/common/PersistentStateManager.ts

```diff
@@ -20,15 +21,16 @@ export class PersistentStateManager {
     defaultState: State | null = null,
     private readonly router: Router | null = null
   ) {
+    this.knownKeys = defaultState ? Object.keys(defaultState) : []
     const storedState = localStorage.getItem(this.getKey())
     if (storedState == null) {
       this.state = defaultState ?? {}
     } else {
       // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-      this.state = JSON.parse(storedState)
+      this.state = this.stateWithKnownKeys(JSON.parse(storedState))
```

Comment:
Please apply `JSON.parse(storedState) as State` instead of `// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment`